iceoryx_utils/include/iceoryx_utils/internal/cxx/forward_list.inl

```diff
@@ -0,0 +1,538 @@
+    }
+
+    const sizeType l_toBeUsedIdx = getHeadFreeEl()->nextIdx;
+
+    if (m_size < CAPACITY && l_toBeUsedIdx < INVALID_INDEX)
+    {
+        // unlink from freeList
+        getHeadFreeEl()->nextIdx = getPointerToListNode()[l_toBeUsedIdx].nextIdx;
+
+        // data class c'tor
+        new (&(getPointerToListNode()[l_toBeUsedIdx].data)) T(std::forward<ConstructorArgs>(args)...);
+
+        // add to usedList
+        const sizeType emplaceAfterIdx = afterToBeEmplacedIter.m_iterListNodeIdx;
+        const sizeType emplaceBeforeIdx = getPointerToListNode()[emplaceAfterIdx].nextIdx;
+        getPointerToListNode()[l_toBeUsedIdx].nextIdx = emplaceBeforeIdx;
+        getPointerToListNode()[emplaceAfterIdx].nextIdx = l_toBeUsedIdx;
+
+        ++m_size;
+
+        return iterator{const_cast<forward_list<T, CAPACITY>*>(afterToBeEmplacedIter.m_cList), l_toBeUsedIdx};
+    }
+    else
+    {
+        errorMessage(__PRETTY_FUNCTION__, " capacity exhausted ");
+        return end();
+    }
+}
+
+
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::iterator
+forward_list<T, CAPACITY>::erase_after(const_iterator beforeToBeErasedIter) noexcept
+{
+    if (this != beforeToBeErasedIter.m_cList)
+    {
+        errorMessage(__PRETTY_FUNCTION__, " iterator of other list can't be used ");
+        std::terminate();
+    }
+
+    sizeType l_beforeToBeRmIdx = beforeToBeErasedIter.m_iterListNodeIdx;
+    if (l_beforeToBeRmIdx >= INVALID_INDEX)
+    {
+        errorMessage(__PRETTY_FUNCTION__, " iterator is end()");
+        return end();
+    }
+
+    sizeType l_toBeRmIdx = getPointerToListNode()[l_beforeToBeRmIdx].nextIdx;
+
+    if (l_toBeRmIdx < INVALID_INDEX && !empty())
+    {
+        // unlink from usedList
+        getPointerToListNode()[l_beforeToBeRmIdx].nextIdx = getPointerToListNode()[l_toBeRmIdx].nextIdx;
+
+        // data'tor data class
+        getPointerToListNode()[l_toBeRmIdx].data.~T();
+
+        // add to freeList
+        getPointerToListNode()[l_toBeRmIdx].nextIdx = getHeadFreeEl()->nextIdx;
+        getHeadFreeEl()->nextIdx = l_toBeRmIdx;
+
+        --m_size;
+
+        // Iterator to the element following the erased one, or end() if no such element exists.
+        return iterator{const_cast<forward_list<T, CAPACITY>*>(beforeToBeErasedIter.m_cList),
+                        getPointerToListNode()[l_beforeToBeRmIdx].nextIdx};
+    }
+
+    // in case list is empty or iterator reached end()
+    return end();
+}
+
+
+template <typename T, int64_t CAPACITY>
+bool forward_list<T, CAPACITY>::push_front(const T& ref) noexcept
+{
+    return emplace_front(ref);
+}
+
+template <typename T, int64_t CAPACITY>
+bool forward_list<T, CAPACITY>::push_front(T&& uref) noexcept
+{
+    return emplace_front(std::forward<T>(uref));
+}
+
+template <typename T, int64_t CAPACITY>
+bool forward_list<T, CAPACITY>::pop_front() noexcept
+{
+    auto sizeBeforeErase = m_size;
+    erase_after(before_begin());
```

Comment:
Is the `front` supposed to be the first element whereas the `begin` is the iterator pointing to it ?